sycl/include/sycl/ext/oneapi/sub_group_mask.hpp

```diff
@@ -249,7 +249,8 @@ struct sub_group_mask {
   }
 
 private:
-  sub_group_mask(BitsType rhs, size_t bn) : Bits(rhs), bits_num(bn) {
+  sub_group_mask(BitsType rhs, size_t bn)
```

Comment:
Note for reviewers: I had to update this constructor or otherwise I faced some issues with comparing smaller masks.

Problematic use case I had:
```
uint32_t all_ones_bitmask = -1;
sub_group_mask all_ones = create(all_ones_bitmask, 16);
sub_group_mask all_zero = create(0, 16);

all_ones.flip();
assert(all_ones == all_zero); // fails!
```

The problem here is that `.flip()` (and other methods) only operate on *actual* (or valuable) bits of the mask, whilst `operator==` compares the whole 32-bit storage. Thus, `all_ones` flipped is actually 16 ones and then 16 zeros, whilst `all_zero` is 32 zeros.